From 360a7faf54c7dafc54434238935170b2f72cda24 Mon Sep 17 00:00:00 2001
From: Xu Mao <B34688@freescale.com>
Date: Wed, 5 Mar 2014 20:12:34 +0800
Subject: [PATCH 46/95] ENGR00133115 use IsSupportedUrl() to check url

revert the modification in mediaplayer.cpp
because it's not necessary

Signed-off-by: Xu Mao <B34688@freescale.com>
---
 include/media/OMXPlayer.h                          |  7 +++++++
 media/libmedia/mediaplayer.cpp                     |  5 -----
 media/libmediaplayerservice/MediaPlayerFactory.cpp | 23 ++++++++--------------
 3 files changed, 15 insertions(+), 20 deletions(-)

diff --git a/include/media/OMXPlayer.h b/include/media/OMXPlayer.h
index dceff34..abd1322 100644
--- a/include/media/OMXPlayer.h
+++ b/include/media/OMXPlayer.h
@@ -115,12 +115,19 @@ private:
     bool                bNetworkFail;
 };
 
+typedef enum {
+    URL_SUPPORT = 0,
+    URL_NOT_SUPPORT,
+    URL_UNKNOWN,
+}URL_TYPE;
+
 class OMXPlayerType
 {
 public:
                         OMXPlayerType();
     virtual             ~OMXPlayerType();
     int                 IsSupportedContent(char *url);
+    URL_TYPE            IsSupportedUrl(const char *url);
 };
 
 }; // namespace android
diff --git a/media/libmedia/mediaplayer.cpp b/media/libmedia/mediaplayer.cpp
index ba816dd..502ab2d 100644
--- a/media/libmedia/mediaplayer.cpp
+++ b/media/libmedia/mediaplayer.cpp
@@ -395,11 +395,6 @@ bool MediaPlayer::isPlaying()
             ALOGE("internal/external state mismatch corrected");
             mCurrentState = MEDIA_PLAYER_STARTED;
         }
-        if ((mCurrentState & MEDIA_PLAYER_PLAYBACK_COMPLETE) && temp) {
-            ALOGE("internal/external state mismatch corrected");
-            mCurrentState = MEDIA_PLAYER_STARTED;
-        }
-        
         return temp;
     }
     ALOGV("isPlaying: no active player");
diff --git a/media/libmediaplayerservice/MediaPlayerFactory.cpp b/media/libmediaplayerservice/MediaPlayerFactory.cpp
index 7227a75..0d1e507 100644
--- a/media/libmediaplayerservice/MediaPlayerFactory.cpp
+++ b/media/libmediaplayerservice/MediaPlayerFactory.cpp
@@ -378,26 +378,19 @@ class OMXPlayerFactory : public MediaPlayerFactory::IFactory {
             if (kOurScore <= curScore)
                 return 0.0;
 
-            if (!strncasecmp("widevine://", url, 11)) {
-                return 0.0;
-            }
-
             if (!strncasecmp(url, "http://", 7)) { 
                 if (isWVM(url, headers))
                 return 0.0;
-
-                size_t len = strlen(url);
-                if (len >= 5 && !strcasecmp(".m3u8", &url[len - 5])) {
-                    return 0.0;
-                }
-                
             }
-
-            if (!strncasecmp(url, "http://", 7) \
-                    || !strncasecmp(url, "rtsp://", 7) \
-                    || !strncasecmp(url, "udp://", 6) \
-                    || !strncasecmp(url, "rtp://", 6))
+            
+            URL_TYPE url_type;
+            OMXPlayerType *pType = new OMXPlayerType();
+            url_type = pType->IsSupportedUrl(url);
+            delete pType;
+            if(url_type == URL_SUPPORT)
                 return kOurScore;
+            else if(url_type == URL_NOT_SUPPORT)
+                return 0.0;
 
             int lenURL = strlen(url);
             for (int i = 0; i < NELEM(FILE_EXTS); ++i) {
-- 
1.8.0

